plugins/core/src/terminal-service.ts

```diff
@@ -0,0 +1,43 @@
+import { ScryptedDeviceBase, StreamService } from "@scrypted/sdk";
+import type { IPty, spawn as ptySpawn } from 'node-pty-prebuilt-multiarch';
+import { BufferedBuffer } from '@scrypted/common/src/buffered-buffer';
+
+export const TerminalServiceNativeId = 'terminalservice';
+
+export class TerminalService extends ScryptedDeviceBase implements StreamService {
+    async connectStream(input: AsyncGenerator<any, void>): Promise<AsyncGenerator<any, void>> {
+        const spawn = require('node-pty-prebuilt-multiarch').spawn as typeof ptySpawn;
+        const cp: IPty = spawn(process.env.SHELL, [], {});
+        const buffer = new BufferedBuffer();
+        buffer.onClose(() => cp.kill());
```

Comment:
This is unnecessary and also racey. on end will be called before the queue is flushed.